### PR TITLE
Make "FunctionStats" a class (rather than a proto)

### PR DESCRIPTION
**Pull request by florian-kuebler**
This also renames "FunctionStats" to "ScopeStats" to prepare for
a more generic usage with manual instrumenation timers.

Test: Compile
Bug: http://226551548

### CODE REVIEW

File: src/ClientData/include/ClientData/ScopeStats.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  [[nodiscard]] double variance_ns() const { return variance_ns_; }
+  void set_variance_ns(double variance_ns) { variance_ns_ = variance_ns; }
+
+  [[nodiscard]] uint64_t std_dev_ns() const { return std_dev_ns_; }
+  void set_std_dev_ns(uint64_t std_dev_ns) { std_dev_ns_ = std_dev_ns; }
+
+ private:
+  uint64_t count_;
+  uint64_t total_time_ns_;
+  uint64_t average_time_ns_;
```
Comment (akopich) [inline src/ClientData/include/ClientData/ScopeStats.h:47]: Consider removing this field, as you have `total_time_ns_` and `count_`

File: src/ClientData/include/ClientData/ScopeStats.h [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  uint64_t min_ns_;
+  uint64_t max_ns_;
+  double variance_ns_;
```
Comment (akopich) [inline src/ClientData/include/ClientData/ScopeStats.h:50]: So the average is `uint64_t`, but the variance is `double`. Is this so for historical reasons?

**Review by akopich — COMMENTED** [review]
Thanks for the change!

File: src/ClientData/include/ClientData/ScopeStats.h (hunk 2 above)
Comment (florian-kuebler) [reply src/ClientData/include/ClientData/ScopeStats.h:50]: Yeah, I think this has historic reasons. We report the average  in the Live tab and show a rounded value there. While the variance was added mostly as internal value and precision matters. We can consider making "average" also double. But I would defer this to another change.

File: src/ClientData/include/ClientData/ScopeStats.h (hunk 1 above)
Comment (florian-kuebler) [reply src/ClientData/include/ClientData/ScopeStats.h]: Good point. Done.

Comment (florian-kuebler) [reply src/ClientData/include/ClientData/ScopeStats.h]: That also let me spot some duplicated code.

**Comment by akopich** [comment]
LGTM

**Review by akopich — APPROVED** [review]

**Pull request merged**